Extensions/Terraform/Src/Tasks/TerraformTaskV1/src/terraform-commands.ts

```diff
@@ -0,0 +1,73 @@
+export class BaseTerraformCommand {
+    public readonly name: string;
+    public readonly workingDirectory: string;
+    public readonly additionalArgs: string | undefined;
+
+    constructor(
+        name: string,
+        workingDirectory: string,
+        additionalArgs?: string | undefined
+    ) {
+        this.name = name;
+        this.workingDirectory = workingDirectory;  
+        this.additionalArgs = additionalArgs;
+    } 
+}
+
+export class TerraformInit extends BaseTerraformCommand {
+    readonly backendType: string | undefined;
+
+    constructor(
+        name: string,
```

Comment:
You have a specific init, plan classes, why do you take the name as input. Aren't they init, plan respectively.